sql/catalog.go

```diff
@@ -51,9 +53,20 @@ func (d Databases) Table(dbName string, tableName string) (Table, error) {
 		return nil, err
 	}
 
+	tableName = strings.ToLower(tableName)
+
 	tables := db.Tables()
-	table, found := tables[tableName]
-	if !found {
+	// Try to get the table by key, but if the name is not the same,
+	// then use the slow path and iterate over all tables comparing
+	// the name.
+	table, ok := tables[tableName]
+	if !ok {
+		for name, table := range tables {
```

Comment:
I thought that we'll store table names only lowercase  (so on `Create` table we'll convert everything to lowercase). It will avoid extra _looping_ every time.

Reply:
We do, other implementations of `Database` may not